src/components/Inngest.ts

```diff
@@ -0,0 +1,164 @@
+import axios, { AxiosInstance, AxiosResponse } from "axios";
+import * as InngestT from "../types";
+import { InngestFunction } from "./InngestFunction";
+import { InngestStep } from "./InngestStep";
+
+/**
+ * A client used to interact with the Inngest API by sending or reacting to
+ * events.
+ *
+ * To provide event typing, make sure to pass in your generated event types as
+ * the first generic.
+ */
+export class Inngest<Events extends Record<string, InngestT.EventPayload>> {
+  /**
+   * The name of this instance, most commonly the name of the application it
+   * resides in.
+   */
+  public readonly name: string;
+
+  /**
+   * Inngest Source API key, used to send events to Inngest Cloud.
+   */
+  private readonly apiKey: string;
+
+  /**
+   * Base URL for Inngest Cloud.
+   */
+  public readonly inngestBaseUrl: URL;
+
+  /**
+   * The URL of the Inngest Cloud API.
+   */
+  private readonly inngestApiUrl: URL;
+
+  /**
+   * An Axios instance used for communicating with Inngest Cloud.
+   *
+   * @link https://npm.im/axios
+   */
+  private readonly client: AxiosInstance;
+
+  constructor(
+    /**
+     * The name of this instance, most commonly the name of the application it
+     * resides in.
+     */
+    name: string,
+
+    /**
+     * Inngest Source API key, used to send events to Inngest Cloud.
+     */
+    apiKey: string,
```

Comment:
Instead of API key, let's call this an `event key`.  API keys are keys to manage your account.